torch/_subclasses/fake_tensor.py

```diff
@@ -856,16 +856,7 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
             with self:
                 return decomposition_table[func](*args, **kwargs)
 
-        # prims already wrap FakeTensor inputs to FakeTensor outputs
```

Comment:
as @ezyang  said
> prims already performed device logic, and so clearing out the device information meant that it no longer works, namely cpu promotion

Ideally, we could rely on the device logic of prims, as is the case today.